[CI] IndexFoldersDeletionListenerIT > testListenersInvokedWhenIndexIsDangling occassionally failing

Looks like a newly added test that has failed a few times now.

**Build scan**:
https://gradle-enterprise.elastic.co/s/wits5fsx622y2/tests/:server:internalClusterTest/org.elasticsearch.plugins.IndexFoldersDeletionListenerIT/testListenersInvokedWhenIndexIsDangling#1

**Repro line**:
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.testListenersInvokedWhenIndexIsDangling" -Dtests.seed=8A30512BA393F582 -Dtests.security.manager=true -Dtests.locale=es -Dtests.timezone=EST -Druntime.java=11`

**Reproduces locally?**:
Nope.

**Applicable branches**:
`master` and `7.x`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P28D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.plugins.IndexFoldersDeletionListenerIT&tests.disabledDistributions=WyJvdXRjb21lOnBhc3NlZCJd&tests.sortField=FAILED&tests.test=testListenersInvokedWhenIndexIsDangling&tests.unstableOnly=true

**Failure excerpt**:
```
org.elasticsearch.plugins.IndexFoldersDeletionListenerIT > testListenersInvokedWhenIndexIsDangling FAILED
    java.lang.AssertionError: Expecting no shards deleted on node [node_t3] but got: {[hokkjihfdt/ldTt4FW9Q6CLZ7hUXlrj4A]=[[hokkjihfdt][7]]}
        at __randomizedtesting.SeedInfo.seed([8A30512BA393F582:60BB557EF8D46F5D]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.assertNoDeletions(IndexFoldersDeletionListenerIT.java:368)
        at org.elasticsearch.plugins.IndexFoldersDeletionListenerIT.testListenersInvokedWhenIndexIsDangling(IndexFoldersDeletionListenerIT.java:214)
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/plugins/IndexFoldersDeletionListenerIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/plugins/IndexFoldersDeletionListenerIT.java
@@ -68,6 +68,15 @@ public class IndexFoldersDeletionListenerIT extends ESIntegTestCase {
         return plugins;
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            // prevent shards to move around after they got assigned the first time
+            .put(EnableAllocationDecider.CLUSTER_ROUTING_REBALANCE_ENABLE_SETTING.getKey(), EnableAllocationDecider.Rebalance.NONE)
+            .build();
+    }
+
     public void testListenersInvokedWhenIndexIsDeleted() throws Exception {
         final String masterNode = internalCluster().startMasterOnlyNode();
         internalCluster().startDataOnlyNodes(2);
@@ -75,8 +84,8 @@ public class IndexFoldersDeletionListenerIT extends ESIntegTestCase {
 
         final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
         createIndex(indexName, Settings.builder()
-            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 2 * between(1, 2))
-            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, between(0, 1))
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 2)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
             .build());
 
         final NumShards numShards = getNumShards(indexName);
@@ -129,7 +138,7 @@ public class IndexFoldersDeletionListenerIT extends ESIntegTestCase {
 
         final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
         createIndex(indexName, Settings.builder()
-            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 4 * between(1, 2))
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 4)
             .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, between(0, 1))
             .build());
 
@@ -192,7 +201,7 @@ public class IndexFoldersDeletionListenerIT extends ESIntegTestCase {
 
         final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
         createIndex(indexName, Settings.builder()
-            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 4 * between(1, 2))
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 4)
             .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, between(0, 1))
             .build());
 

